### ISSUE TITLE
firestore: pagination using query cursor on subcollection not working

### ISSUE THREAD

**Issue by learncode** [googleapis/google-cloud-go#1659]
While trying to paginate on a subcollection, when passing a valid document I come across the following error [`firestore: document snapshot for %s passed to query on %s `](https://github.com/googleapis/google-cloud-go/blob/master/firestore/query.go#L406). 

Exact error: `firestore: document snapshot for projects/project-name/databases/(default)/documents/parent/123/child passed to query on projects/project-name/databases/(default)`

```
query := f.client.CollectionGroup(childCollection).
	Where("RecordedAt", ">=", request.FromDate).
	Where("RecordedAt", "<=", request.ToDate).
	Where("RequestID", "==", requestID)

if request.Paging.LastDocumentID != "" {
	iter := f.client.CollectionGroup(childCollection).
		Where("ID", "==", request.Paging.LastDocumentID).
		Documents(ctx)
	doc, err := iter.Next()
	if err != nil {
		return err
	}
	query = query.StartAfter(doc)
}

query = query.Limit(request.Paging.PageSize).OrderBy("RecordedAt", firestore.Desc)

//Fails here
docs, err := query.Documents(ctx).GetAll()
if err != nil {
	return response, err
}
```

The case when LastDocumentID is empty, records are fetched fine. The error scenario is when passing a document, that is fetched based on the LastDocumentID. Just to add paginating on the parent I do not encounter any such issues. Is there something I'm doing incorrectly, please let me know.

**Comment by BenWhitehead** [comment]
Possibly related to #1225 but I'm not certain as to why.

@learncode What version of the firestore library and what version of go are you using?

**Comment by learncode** [comment]
I'm using version `1.1.0`.  

> cloud.google.com/go/firestore v1.1.0

**Comment by tritone** [comment]
This should contain the fix to #1225 then. I will mark it as a bug and investigate further.

**Comment by learncode** [comment]
@tritone any update on this. We are currently blocked on delivering a feature. Really appreciate if you can quickly take a look.

**Comment by learncode** [comment]
@tritone & @BenWhitehead could you please let me know if there is any update on this issue. Really appreciate if you can quickly take a look.

**Comment by tritone** [comment]
Hi @learncode , apologies, I was traveling last week. Will take a look today.

**Comment by tritone** [comment]
Hi @learncode , I took a closer look today and this does look like a bug. I found an issue in the python client library that looks very similar ( https://github.com/googleapis/google-cloud-python/issues/8633 ) as well as a PR that fixes the issue. Basically it looks like CollectionGroup queries should be excepted from the validation error you encountered. I will make a PR to fix this.

**Comment by learncode** [comment]
@tritone thank you for taking a look. is there a timeline on when can the fix be pushed. Thanks again for all the help

**Comment by tritone** [comment]
I have https://code-review.googlesource.com/c/gocloud/+/49150 up for review, please try this out to see if it resolves your issue if you have time. it will probably be a week or so to get through review and release the firestore client with this fix.

**Comment by tritone** [comment]
Released in firestore/1.1.1
